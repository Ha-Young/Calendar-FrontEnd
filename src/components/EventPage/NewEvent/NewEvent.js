import React from "react";
import { FORM_ID } from "../../../constants/common";
import Form from "../../../containers/EventContainer/FromContainer";
import styles from "./NewEvent.module.css";
import PropTypes from "prop-types";

function NewEvent() {
  return (
    <>
      <h1>new event</h1>
      <div className={styles.eventContainer}>
        <Form formId={FORM_ID.ADD}/>
        <button type='submit' form={FORM_ID.ADD}>Add Event</button>
      </div>
    </>
  );
}

export default NewEvent;
