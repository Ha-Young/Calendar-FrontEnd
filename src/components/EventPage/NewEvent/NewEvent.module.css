.eventContainer {
  margin-top: 10%;
}

.eventContainer form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.eventContainer button {
  width: 384px;
  margin-top: 10px;
  padding: 10px;
  font-size: 20px;
  background-color: inherit;
  color: inherit;
  border-radius: 5px;
  cursor: pointer;
  outline: none;
}
