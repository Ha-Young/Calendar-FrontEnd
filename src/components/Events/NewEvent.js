import React, { useState } from "react";
import Form from "./Form";
import styles from "./Events.module.css";
import { useLocation } from "react-router-dom";
import { getOnlyHours, makeOClock } from "../../utils/event";
import { deleteTargetData, writeUserData } from "../../api";

export default function NewEvent({ userId }) {
  const location = useLocation();

  let initialInputValues = {
    date: "",
    title: "",
    detail: "",
    startAt: "",
    endAt: "",
  };

  if (location.state) {
    initialInputValues = {
      ...location.state,
      startAt: makeOClock(location.state.startAt),
      endAt: makeOClock(location.state.endAt),
    };
  }

  const [inputValues, setInputValues] = useState(initialInputValues);

  function onSubmit(e) {
    e.preventDefault();

    const {
      date, 
      title,
      detail,
      startAt,
      endAt,
    } = inputValues;

    if (location.state) {
      deleteTargetData(
        userId, 
        initialInputValues.date,
        getOnlyHours(initialInputValues.startAt), 
        getOnlyHours(initialInputValues.endAt)
      );
    }

    writeUserData(userId, date, title, detail, getOnlyHours(startAt), getOnlyHours(endAt));
  }

  function onChangeInputValues(e) {
    let { value, name } = e.target;

    if (name === "startAt" || name === "endAt") {
      value = makeOClock(value);
    }

    setInputValues(prev => {
      return {
        ...prev,
        [name]: value,
      }
    })
  }

  return (
    <div className={`${styles.eventFormContainer} ${styles.flexCenter}`}>
      <Form
        inputValues={inputValues}
        onChangeInputValues={onChangeInputValues}
        onSubmit={onSubmit}
      />
    </div>
  );
}
