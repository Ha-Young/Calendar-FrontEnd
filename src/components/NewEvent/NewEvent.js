import React, { useState, useEffect } from "react";
import { useHistory } from "react-router-dom";
import moment from "moment";
import styles from "./NewEvent.module.css";
import Modal from "../Modal/Modal";

export default function NewEvent ({ addEvent, errorMessage }) {
  const history = useHistory();
  const [eventDetails, setEventDetails] = useState({
    eventName: "",
    eventDescription: "",
    eventDate: "",
    startTime: "",
    endTime: "",
  });
  const {
    eventName,
    eventDescription,
    eventDate,
    startTime,
    endTime
  } = eventDetails;

  const handleSubmit = (event) => {
    event.preventDefault();
  };

  const handleInputsChange = (event) => {
    const { value, name } = event.target;
    setEventDetails({
      ...eventDetails,
      [name]: value,
    });
  };

  const handleSubmitButton = async () => {
    const id = "event" + Date.now();
    await addEvent({
      id: id,
      createdAt: moment().toISOString(),
      ...eventDetails,
    });

    history.push("/calendar");
  };

  const handleCancelButton = (event) => {
    event.preventDefault();
    history.push("/calendar");
  };

  const isEventDetailsFilled = eventName && eventDescription && eventDate && startTime && endTime;

  return (
    <>
      {
        errorMessage
        && <Modal title={errorMessage} />
      }
      <div className={styles.NewEvent}>
        <h2>Create Events</h2>
        <form onSubmit={handleSubmit}>
          <label htmlFor="eventName">Event Name</label>
          <input
            type="text"
            name="eventName"
            placeholder="Event Name"
            value={eventName}
            onChange={handleInputsChange}
            required
          />
          <label htmlFor="eventDescription">Event Description</label>
          <input
            type="text"
            name="eventDescription"
            placeholder="Event Description"
            value={eventDescription}
            onChange={handleInputsChange}
            required
          />
          <label htmlFor="eventDate">Event Date</label>
          <input
            type="date"
            name="eventDate"
            value={eventDate}
            onChange={handleInputsChange}
            required
          />
          <input
            type="time"
            name="startTime"
            value={startTime}
            onChange={handleInputsChange}
            required
          />
          <input
            type="time"
            name="endTime"
            value={endTime}
            onChange={handleInputsChange}
            required
          />
          <button
            className={styles.submitButton}
            onClick={handleSubmitButton}
            disabled={isEventDetailsFilled ? false : true}
          >
            Submit
          </button>
          <button
            className={styles.cancelButton}
            onClick={handleCancelButton}
          >
            Cancel
          </button>
        </form>
      </div>
    </>
  );
}
