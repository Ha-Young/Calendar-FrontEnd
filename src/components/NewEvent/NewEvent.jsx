import { dateConst } from "constants/constants";
import React, { useState } from "react";

// import styles from "./NewEvent.module.css";

const NewEvent = (props) => {
  const [event, setEvent] = useState({});

  const onSubmit = (event) => {
    event.preventDefault();
    console.log(event.target.value);
  };

  const onChange = (e) => {
    const key = e.target.name;
    const value = e.target.value;
    setEvent({ ...event, [key]: value });
    console.log(event);
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        name="title"
        type="text"
        placeholder="Event Title"
        onChange={onChange}
        required
      />
      <input
        name="description"
        type="textarea"
        placeholder="Event Description"
        onChange={onChange}
        required
      />
      <input
        name="date"
        type="date"
        // value={date}
        onChange={onChange}
        required
      />
      <select name="startTime" id="start-time" onChange={onChange} required>
        {dateConst.TIME_LIST.map((time) => (
          <option key={time} value={time}>
            {time}시
          </option>
        ))}
      </select>
      <select name="endTime" id="end-time" onChange={onChange} required>
        {dateConst.TIME_LIST.map((time) => (
          <option key={time} value={time}>
            {time}시
          </option>
        ))}
      </select>
      <button>Add</button>
    </form>
  );
};

export default NewEvent;
