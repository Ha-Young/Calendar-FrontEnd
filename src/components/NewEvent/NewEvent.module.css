.NewEvent {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: stretch;
  height: 100vh;
  padding: 100px 100px;
}

.NewEvent h2 {
  color: var(--main-color);
}

.NewEvent form {
  display: flex;
  flex-direction: column;
}

.NewEvent form * {
  width: 300px;
}

.NewEvent form input {
  height: 30px;
  border: none;
  border-bottom: 1px solid var(--main-color);
}

.NewEvent form label {
  margin: 30px 0 3px;
  color: var(--main-color);
  font-size: 14px;
}

.submitButton {
  height: 40px;
  margin-top: 30px;
  border: none;
  border-radius: 20px;
  background-color: var(--main-color);
  color: white;
  text-transform: uppercase;
}

.submitButton:disabled {
  height: 40px;
  margin-top: 30px;
  border: none;
  border-radius: 20px;
  background-color: rgba(0, 0, 0, 0.2);
  color: white;
  text-transform: uppercase;
}

.cancelButton {
  height: 40px;
  margin-top: 10px;
  border: none;
  border-radius: 20px;
  background-color: rgba(0, 0, 0, 0.2);
  color: white;
  text-transform: uppercase;
}
