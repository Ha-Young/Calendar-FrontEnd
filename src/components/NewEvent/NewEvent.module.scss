.NewEvent {
  width: 60%;
  display: flex;
  flex-direction: column;
  margin: 0 auto;

  h2 {
    text-align: left;
  }
}
