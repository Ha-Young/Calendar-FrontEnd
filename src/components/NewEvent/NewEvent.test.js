import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import NewEvent from "./NewEvent";

afterEach(cleanup);

it("matches snapshot", () => {
  const utils = render(<NewEvent />);
  expect(utils.container).toMatchSnapshot();
});

it("enables button only when all inputs are entered", () => {
  const { getByText, getByLabelText } = render(<NewEvent />);

  const submitButton = getByText("Submit");
  const eventName = getByLabelText("Event Date");
  const eventDescription = getByLabelText("Event Description");
  const eventDate = getByLabelText("Event Date");

  expect(submitButton).toBeDisabled();

  fireEvent.change(eventName, { target: { value: "Go home!" } });
  fireEvent.change(eventDescription, { target: { value: "Go home!" } });
  fireEvent.change(eventDate, { target: { value: "2020-10-03" } });

  expect(submitButton).toBeDisabled();
});
