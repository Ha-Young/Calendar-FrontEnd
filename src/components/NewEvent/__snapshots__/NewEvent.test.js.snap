// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="NewEvent"
  >
    <h2>
      Create Events
    </h2>
    <form>
      <label
        for="eventName"
      >
        Event Name
      </label>
      <input
        id="eventName"
        placeholder="Event Name"
        required=""
        type="text"
        value=""
      />
      <label
        for="eventDescription"
      >
        Event Description
      </label>
      <input
        id="eventDescription"
        placeholder="Event Description"
        required=""
        type="text"
        value=""
      />
      <label
        for="eventDate"
      >
        Event Date
      </label>
      <input
        id="eventDate"
        required=""
        type="date"
        value=""
      />
      <input
        id="startTime"
        required=""
        type="time"
        value=""
      />
      <input
        id="endTime"
        required=""
        type="time"
        value=""
      />
      <button
        class="submitButton"
        disabled=""
      >
        Submit
      </button>
      <button
        class="cancelButton"
      >
        Cancel
      </button>
    </form>
  </div>
</div>
`;
